### PR TITLE
Create scaleContactSpheres.m

### PR DESCRIPTION
**Pull request by VDB-Bram**
New function to automatically scale some values of the contact spheres, see #81

### CODE REVIEW

**Comment by VDB-Bram** [comment]
@InesVandekerckhove, thanks for taking a look and testing! Good to hear it works without errors :) 

What is your reasoning behind taking the mean of both scaling factors?

**Comment by dgupta7** [comment]
It is not clear where this function is to be used and how the scaling values of stiffness and damping are calculated (I think there is a separate place of those calculations?). It is needed to be added for this function to be useable.

**Comment by VDB-Bram** [comment]
> It is not clear where this function is to be used and how the scaling values of stiffness and damping are calculated (I think there is a separate place of those calculations?). It is needed to be added for this function to be useable.

I'll see where a description of this function fits in the readme. The current code does not calculate stiffness and dissipation, it only implements prescribed scaling factors. I.e., the user chooses which scaling factors to use.

**Comment by miriamfebrer** [comment]
Hi @VDB-Bram , I have tested the function and it works correctly regarding the scaling of the radius, stiffness and dissipation.

But I have two comments, in the same line as @dgupta7 .

1. After running this function, you should also run 'fixContactSpherePositionAfterScaling.m' to locate correctly the spheres. And maybe it would make sense to merge both functions, or there are cases where you only want to move the spheres, without scaling? Then, instead of merging, we can call 'fixContactSpherePositionAfterScaling.m' at the end of 'scaleContactSpheres.m'?

2. Would it make sense to have an 'Example_contact_spheres_scaling.m' to show how to use these functions? Something like this:
```
genericModelPath = 'C:\Example\Falisse_et_al_2022.osim';
modelpath_in = 'C:\Example\Falisse_et_al_2022_scaled.osim';
modelout_path = 'C:\Example\Falisse_et_al_2022_scaled_CS.osim';
scale.stiffness = mass_child/mass_generic*(height_generic/height_child)^2;
scale.dissipation = height_child/height_generic;
scaleContactSpheres(genericModelPath,modelpath_in,modelout_path,scale);
scaledModelPath = modelout_path;
fixContactSpherePositionAfterScaling(genericModelPath,scaledModelPath); %,varargin)
```

**Comment by VDB-Bram** [comment]
@miriamfebrer, thanks for your comments. I instead implemented it into AdaptOpenSimModel.m. You can choose to do the scaling or not by stetting the booleans.

Is was also now wondering if it would be useful to also allow a scale.radius as input instead of scaling the spheres based on the foot length. It is possible to keep that functionality though in case scale.radius is empty.

**Comment by VDB-Bram** [comment]
@miriamfebrer, current PR is up to date with main. If this PR is ok, can you give the approval?

File: AdaptOpenSimModel/AdaptOpenSimModel.m [hunk 1]
```diff
@@ -48,6 +48,7 @@
 add_actuators_bool = 1;
 add_contact_bool = 0;
 use_reference_contacts_bool = 1;
+scale_contact_spheres_bool = 1;
```
Comment (miriamfebrer) [inline AdaptOpenSimModel/AdaptOpenSimModel.m:51]: You should add the description of this new setting on the header

File: AdaptOpenSimModel/scaleContactSpheres.m [hunk 2]
```diff
@@ -0,0 +1,99 @@
+function scaleContactSpheres(genericModelPath,modelpath_in,modelout_path,scale)
+% This code scales:
+% 1. the radius of the contact spheres based on the scaling factor of the
+% foot. 
+% 2. the stiffness of the contactsphere
```
Comment (miriamfebrer) [inline AdaptOpenSimModel/scaleContactSpheres.m:5]: small edit: space between "contact" and "sphere"

**Pull request merged**